tests/semantic_analyser.cpp

```diff
@@ -944,7 +935,7 @@ kprobe:f { @x = 1; delete(@x) ? 0 : 1; }
 )");
 
   test_error("kprobe:f { @y[5] = 5; delete(@y[5], 5); }", R"(
-stdin:1:23-35: ERROR: delete() expects a map with no keys for the first argument
+stdin:1:23-35: ERROR: delete() expects a map argument
```

Comment:
This error messaging still isn't great. I won't block but we should address this in a follow up PR. It should say "delete() expects a map argument. Got a map access" or something like that.